bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -675,19 +675,42 @@ type DiskSetup struct {
 	Filesystems []Filesystem `json:"filesystems,omitempty"`
 }
 
+// DiskLayout represents an array of partition specifications
+type DiskLayout []PartitionSpec
+
+// PartitionSpec defines the size and optional type for a partition
+type PartitionSpec struct {
+	// Percentage of disk that partition will take (1-100)
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=100
+	Percentage int32 `json:"percentage"`
+
+	// PartitionType is the numerical value of the partition type (optional)
+	// +optional
+	PartitionType *int32 `json:"partitionType,omitempty"`
+}
+
 // Partition defines how to create and layout a partition.
 type Partition struct {
 	// device is the name of the device.
 	Device string `json:"device"`
+
 	// layout specifies the device layout.
 	// If it is true, a single partition will be created for the entire device.
 	// When layout is false, it means don't partition or ignore existing partitioning.
 	Layout bool `json:"layout"`
+
+	// diskLayout specifies the percentage of disk space and partition types.
```

Comment:
Is this list ordered?

Can you validate that the percentages sum to 100? Are they allowed to sum to less than 100?